File: ref/misc/DumpC.java
// Dump all functions decompiled to a text file (Java)
// @category Analysis
import ghidra.app.decompiler.DecompInterface;
import ghidra.app.decompiler.DecompileResults;
import ghidra.util.task.ConsoleTaskMonitor;
import ghidra.program.model.listing.Function;
import ghidra.program.model.listing.FunctionIterator;
import java.io.FileWriter;

public class DumpC extends GhidraScript {
    @Override
    public void run() throws Exception {
        DecompInterface ifc = new DecompInterface();
        ifc.openProgram(currentProgram);
        StringBuilder out = new StringBuilder();
        FunctionIterator funcs = currentProgram.getFunctionManager().getFunctions(true);
        int n = 0;
        for (Function f : funcs) {
            DecompileResults res = ifc.decompileFunction(f, 60, new ConsoleTaskMonitor());
            if (res != null && res.decompileCompleted()) {
                out.append("//===== ").append(f.getName()).append(" @ 0x")
                   .append(Long.toHexString(f.getEntryPoint().getOffset())).append(" =====\n")
                   .append(res.getDecompiledFunction().getC()).append("\n\n");
                n++;
            }
        }
        String name = currentProgram.getName().replace("/", "_");
        String path = "/tmp/bioswork/decomp_" + name + ".c";
        FileWriter fw = new FileWriter(path);
        fw.write(out.toString());
        fw.close();
        println("dumped " + n + " functions to " + path);
    }
}

File: ref/misc/dumpc.py
# Dump all functions decompiled to a text file
# @category Analysis
from ghidra.app.decompiler import DecompInterface
from ghidra.util.task import ConsoleTaskMonitor

prog = getCurrentProgram()
ifc = DecompInterface()
ifc.openProgram(prog)

out = []
fm = prog.getFunctionManager()
funcs = fm.getFunctions(True)
for f in funcs:
    res = ifc.decompileFunction(f, 60, ConsoleTaskMonitor())
    if res and res.decompileCompleted():
        out.append("//===== %s @ 0x%x =====\n%s" % (f.getName(), f.getEntryPoint().getOffset(), res.getDecompiledFunction().getC()))

with open("/tmp/bioswork/decomp_%s.c" % prog.getName(), "w") as fp:
    fp.write("\n\n".join(out))
print("dumped %d functions" % len(funcs))